app/src/lib/menu-update.ts

```diff
@@ -145,8 +145,8 @@ function getMenuState(state: IAppState): Map<MenuIDs, IMenuItemState> {
     menuStateBuilder.setEnabled('compare-branch', isHostedOnGitHub && hasPublishedBranch)
 
     menuStateBuilder.setEnabled('view-repository-on-github', isHostedOnGitHub)
-    menuStateBuilder.setEnabled('push', !networkActionInProgress)
-    menuStateBuilder.setEnabled('pull', !networkActionInProgress)
+    menuStateBuilder.setEnabled('push', hasPublishedBranch && !networkActionInProgress)
```

Comment:
This doesn't seem quite right... if the branch hasn't been published yet, they should be able to push it to publish it.